### PR TITLE
PAY-3649 PBA code refactoring changes.

### PR DESCRIPTION
**Pull request by aravindan89**
### JIRA link (if applicable) ###
PAY-3649

### Change description ###
PBA code refactoring changes.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: dependency-check-suppressions.xml [hunk 1]
```diff
@@ -55,6 +55,7 @@
         <cve>CVE-2017-18640</cve>
         <cve>CVE-2019-17638</cve>
         <cve>CVE-2020-24616</cve>
+        <cve>CVE-2020-5421</cve>
```
Comment (AyanDuttaHMCTS) [inline dependency-check-suppressions.xml:58]: Add a note mentioning why we are suppressing this CVE

**Review by AyanDuttaHMCTS — APPROVED** [review]

File: dependency-check-suppressions.xml (hunk 1 above)
Comment (aravindan89) [reply dependency-check-suppressions.xml:58]: @AyanDuttaHMCTS - I am trying to fix it. Hopefully i can remove this CVE. If not i will definitely add a note.

**Review by YasharRahvar — COMMENTED** [review]
It looks to me we are doing too much in controller layer. We need to move some of the logic into service layer.

**Review by YasharRahvar — APPROVED** [review]

File: dependency-check-suppressions.xml (hunk 1 above)
Comment (aravindan89) [reply dependency-check-suppressions.xml:61]: Added notes

**Pull request merged**